templates/s3/s3.py

```diff
@@ -699,16 +700,23 @@ def _monitor_minio(self):
             self.state.check('actions', 'install', 'ok')
         except StateCheckError:
             return
-        vm_robot, _ = self._vm_robot_and_ip()
-        minio = vm_robot.services.get(template_uid=MINIO_TEMPLATE_UID, name=self.guid)
-        state = minio.state
 
-        try:
-            disk_state = state.get('vm', 'disk')
-            self.state.set('vm', 'disk', disk_state['disk'])
-        except:
-            # probably no state set on the minio disk
-            pass
+        def get_state():
+            retries = 3
+            for _ in range(retries):
+                try:
+                    vm_robot, _ = self._vm_robot_and_ip()
```

Comment:
if the robot is not reachable, it will give a `ConnectionError` not `HTTPError` which means the except wont catch it.

Reply:
This is to handle robot failure, when it's reachable but not working properly because disk failure. While we indeed need to also handle connection error, a connection error might be due to your internet is broken. unlike HTTPError where u sure you can reach the node but it' not behaving.